File: backend/src/main/java/com/dpd/backend/controllers/user/UserDTO.java
package com.dpd.backend.controllers.user;

import com.dpd.backend.controllers.address.AddressDTO;
import com.dpd.backend.controllers.phone.PhoneDTO;
import lombok.Getter;
import lombok.Setter;
import org.springframework.format.annotation.DateTimeFormat;

import javax.validation.constraints.Email;
import javax.validation.constraints.Past;
import java.time.LocalDate;
import java.util.List;
import java.util.UUID;

@Getter
@Setter
public class UserDTO {

    private UUID id;
    @Email(message = "Invalid email")
    private String email;
    private String firstName;
    private String lastName;
    private String mothersName;
    private String taxId;
    private String socialSecurityNumber;
    @Past(message = "Date have to be in the past")
    @DateTimeFormat(iso = DateTimeFormat.ISO.DATE)
    private LocalDate dateOfBirth;
    private String placeOfBirth;
    private List<AddressDTO> addresses;
    private List<PhoneDTO> phones;
    
}

File: frontend/src/DTOs/UserDTO.ts
import { AddressDTO } from './AddressDTO';
import { PhoneDTO } from './PhoneDTO';

export interface UserDTO {
    id: string;
    email: string;
    firstName: string | null;
    lastName: string | null;
    mothersName: string | null;
    taxId: string | null;
    socialSecurityNumber: string | null;
    dateOfBirth: Date;
    placeOfBirth: string | null;
    addresses: AddressDTO[];
    phones: PhoneDTO[];
}
